libzecale/circuits/pairing/bls12_377_pairing.tcc

```diff
@@ -105,420 +194,484 @@ template<typename ppT>
 bls12_377_ate_dbl_gadget<ppT>::bls12_377_ate_dbl_gadget(
     libsnark::protoboard<FqT> &pb,
     const bls12_377_G2_proj<ppT> &R,
+    const bls12_377_G2_proj<ppT> &out_R,
+    const bls12_377_ate_ell_coeffs<ppT> &coeffs,
     const std::string &annotation_prefix)
     : libsnark::gadget<libff::Fr<ppT>>(pb, annotation_prefix)
-    , in_R(R)
-    , out_R(pb, " R")
+    , _in_R(R)
+    , _out_R(out_R)
+    , _out_coeffs(coeffs)
 
     // A = Rx * Ry / 2
-    , A(pb, FMT(annotation_prefix, " A"))
-    , check_A(
-          pb, in_R.X, in_R.Y, A * FqT(2), FMT(annotation_prefix, " check_A"))
+    , _A(pb,
+         _in_R.X,
+         _in_R.Y * FqT(2).inverse(),
+         Fqe_variable<ppT>(pb, FMT(annotation_prefix, " Rx*Ry/2")),
+         FMT(annotation_prefix, " _A"))
 
     // B = Ry^2
-    , B(pb, FMT(annotation_prefix, " B"))
-    , check_B(pb, in_R.Y, B, FMT(annotation_prefix, " check_B"))
+    , _B(pb,
```

Comment:
Another place where https://github.com/clearmatics/zecale/issues/20 would apply.

 - `_B` ->`compute_B` (same for the annotation)
 - `Ry^2` (the output wire annotation) -> `B`

Same applies for the other gadgets in this file

Reply:
This should be covered in a later change.